storage/tables/data-types/index.md

```diff
@@ -1,1124 +1,1170 @@
+
+For this reason, please make sure that your primary key columns are not nullable. This is most relevant in CTAS queries, where columns are nullable by default. To work around this, specify the columns as part of the CTAS expression. For example:
+
+```sql
+CREATE TABLE "ctas_table" (
+    "id" NUMBER NOT NULL,
+    "name" VARCHAR(255) NOT NULL,
+    "created_at" TIMESTAMP_NTZ NOT NULL
+) AS SELECT * FROM "typed_table";
+```
+
+### Pros and Cons
+
+- **Pros**
+  - Loading into a workspace is significantly faster than loading into a table without native data types. Casting data is not necessary when loading into a workspace.
+  - A table accessed in a workspace via the [read-only input mapping](https://help.keboola.com/transformations/workspace/#read-only-input-mapping) already has typed columns.
+  - Data types are strictly enforced, ensuring that data in a specific column (like a number column) is consistent with its type.
+- **Cons**
+  - Changing a column type is complicated; see [How to Change Column Types](#changing-types-of-existing-typed-columns).
+  - Keboola does not perform any type conversion during loading. Your data must exactly match the column type in the table in Storage.
+  - Any load of data with incompatible types will fail.
+  - The filtering option in the input mapping section is unavailable for tables with defined data types. If filtering is crucial for your workflow, consider using SQL, Python, or even no-code transformations to filter the data and create a new filtered table.
+
+
+### How to Create a Typed Table Based on a Non-typed Table
+
+For example, we have a non-typed table `non_typed_table` with the following definition:
+
+{: .image-popup}
+![Non-typed table schema](/storage/tables/data-types/non-typed-table-schema.png)
+
+with sample data:
+
+{: .image-popup}
+![Sample data](/storage/tables/data-types/sample-data.png)
+
+To create typed table based on `non_typed_table` create new transformation, choose table input mapping `non_typed_table` and choose table output mapping `typed_table`. The output table must not exist, otherwise it will not become a typed table.
+
+{: .image-popup}
+![Create transformation](/storage/tables/data-types/create-transformation.png)
+
+In a queries section add sql query transforming columns types:
+
+```sql
+CREATE TABLE "typed_table" AS
+    SELECT 
+        CAST(ntt."id" AS VARCHAR(64)) AS "id",
+        CAST(ntt."id_profile" AS INTEGER) AS "id_profile",
+        TO_TIMESTAMP(ntt."date", 'DD.MM.YYYY"T"HH24:MI:SS') AS "date",
+        CAST(ntt."amount" AS INTEGER) AS "amount"
+    FROM "non_typed_table" AS ntt;
+```
+
+Run the transformation and wait until it finishes.
+
+Note when creating a new table, the link from the transformation works even after refreshing the page. Or you can go to the storage and find the table there.
```

Comment:
```suggestion
```

Reply:
top bych zrusil. To je 1. vlastnost UI, ktera nema nic moc co delat s tim co popisujes 2. muze se imho docela rychle menit